src/views/admin/KlasseCreationView.vue

```diff
@@ -25,9 +28,53 @@
 
   const { t }: Composer = useI18n({ useScope: 'global' });
   const router: Router = useRouter();
+  const authStore: AuthStore = useAuthStore();
+  const personStore: PersonStore = usePersonStore();
   const organisationStore: OrganisationStore = useOrganisationStore();
+  // get personenuebersicht for the current user, if we don't have it
+  watch(
+    () => ({
+      uebersicht: personStore.personenuebersicht,
+      currentUser: authStore.currentUser,
+    }),
+    async ({
+      uebersicht,
+      currentUser,
+    }: {
+      uebersicht: PersonStore['personenuebersicht'];
+      currentUser: AuthStore['currentUser'];
+    }) => {
+      if (!(currentUser && currentUser.personId)) return;
+      if (uebersicht && uebersicht.personId === currentUser.personId) return;
 
-  const hasAutoselectedSchule: Ref<boolean> = ref(false);
+      await personStore.getPersonenuebersichtById(currentUser.personId);
+    },
+    { immediate: true },
+  );
+
+  // is set if user has KLASSE_VERWALTEN at exactly one schule
+  const autoselectedSchuleId: ComputedRef<string | null> = computed(() => {
+    if (!authStore.currentUser?.personenkontexte) return null;
+    const pksWithKlasseVerwaltenPermission: Array<PersonenkontextRolleFieldsResponse> =
+      authStore.currentUser.personenkontexte.filter(
+        (personenkontextRolleFieldsResponse: PersonenkontextRolleFieldsResponse) =>
+          personenkontextRolleFieldsResponse.rolle.systemrechte.includes(RollenSystemRecht.KlassenVerwalten),
+      );
+    if (pksWithKlasseVerwaltenPermission.length > 1) return null;
+    const schulenWithKlasseVerwaltenPermission: Array<PersonenkontextRolleFieldsResponse> =
+      pksWithKlasseVerwaltenPermission.filter((personenkontext: PersonenkontextRolleFieldsResponse) => {
+        const schulZuordnung: DBiamPersonenzuordnungResponse | undefined =
+          personStore.personenuebersicht?.zuordnungen.find(
+            (z: DBiamPersonenzuordnungResponse) =>
+              z.sskId === personenkontext.organisationsId && z.typ === OrganisationsTyp.Schule,
+          );
+        return schulZuordnung !== undefined;
+      });
+    if (schulenWithKlasseVerwaltenPermission.length === 1)
+      return schulenWithKlasseVerwaltenPermission[0]!.organisationsId;
+    return null;
+  });
```

Comment:
Wir können vielleicht hier stattdessen ein .reduce() anwenden damit wir nur einmal iterieren müssen:

```suggestion
const autoselectedSchuleId: ComputedRef<string | null> = computed(() => {
  const user = authStore.currentUser;
  if (!user?.personenkontexte) return null;

  // Find all contexts where the user has KlassenVerwalten
  const schulenWithKlasseVerwaltenPermission = user.personenkontexte.reduce(
    (acc: string[], personenkontext) => {
      if (!personenkontext.rolle.systemrechte.includes(RollenSystemRecht.KlassenVerwalten)) return acc;

      const schulZuordnung = personStore.personenuebersicht?.zuordnungen?.find(
        (z) => z.sskId === personenkontext.organisationsId && z.typ === OrganisationsTyp.Schule
      );

      if (schulZuordnung) acc.push(personenkontext.organisationsId);
      return acc;
    },
    []
  );

  return schulenWithKlasseVerwaltenPermission.length === 1 ? schulenWithKlasseVerwaltenPermission[0] : null;
});

```

Reply:
Habe ich übernommen. Ist so auch viel leserlicher :)